pinot-integration-tests/src/test/java/org/apache/pinot/compat/tests/SegmentOp.java

```diff
@@ -241,6 +242,26 @@ private boolean verifySegmentInState(String state)
     return true;
   }
 
+  private boolean verifyRoutingTableUpdated()
```

Comment:
You should really be getting the number of rows before adding the segment, and then the number of rows after adding the segment and make sure that it has increased by the number of rows in the segment. If you do this, then you can remove the other check of segment state model.online.

You can mark this as TODO and take it on in a separate PR

Reply:
marked as TODO.